synphot/wrapper.py

```diff
@@ -0,0 +1,153 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""This module is a wrapper for synphot's most commonly used functions.
+
+"""
+
+# STDLIB
+import math
+import warnings
+
+# THIRD-PARTY
+import numpy as np
+
+# ASTROPY
+from astropy import log
+from astropy import units as u
+from astropy.utils.exceptions import (AstropyUserWarning,
+                                      AstropyDeprecationWarning)
+from astropy.modeling.models import Const1D
+
+# LOCAL
+from . import binning, exceptions, units, utils
+from .models import Empirical1D
+from .spectrum import BaseSourceSpectrum, SourceSpectrum, SpectralElement
+from .observation import Observation, ccd_snr, exptime_from_ccd_snr
+
+
+class Observe:
+    """
+    Parameters
+    ----------
+    source_spec : `~synphot.spectrum.SourceSpectrum`
+        The spectrum of the target object. 
+        If distance is given, the spectrum is assumed to be at the source. 
+    distance : `~astropy.units.quantity.Quantity`
+        Distance to the source spectrum. Default is None.
+        If None, source_spec is assumed to be scaled by the distance.
+        If distance is given, the spectrum is assumed to be at the source.
+    diameter : `~astropy.units.quantity.Quantity`
+        Diameter of the telescope aperture.
+    bandpass : str, list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values, or 
+        `~synphot.spectrum.SpectralElement`
+        The bandpass(es) of this observation. Can either be specified by a 
+        string or a list with dimensions 2 x n, where one row is the
+        list of wavelengths associated with the bandpass, and the other
+        row contains the fractional values of the bandpass.
+        Default is a constant model with amplitude = 1, i.e. as if there
+        were no bandpass in effect.
+    quantum_efficiency : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the quantum efficiency of the instrument's CCD. 
+        Default is None.
+    atmospheric_transmission : list with dimension 2 x n containing
+        `~astropy.units.quantity.Quantity` values
+        A model for the atmopsheric transmission.
+    """
+    @u.quantity_input(diameter=u.m, distance=u.pc, exptime=u.s)
+    def __init__(self, source_spec, diameter, distance=0 * u.pc,
+                 bandpass=SpectralElement(Const1D, amplitude=1),
+                 quantum_efficiency=None, atmospheric_transmission=None,
```

Comment:
I think this might be controversial (👀 @eteq), but I might vote for atmospheric transmission to take a _boolean_ input, and if `True`, return the skycalc transmittance curve with some default parameters; if `False`, don't include atmospheric transmission. I think for 90% of users that will give them the answer they want within a factor of 2 with as little data wrangling as possible.

Reply:
I might even vote the same thing for `quantum_efficiency`. Perhaps it could instead be `silicon_detector=True`, and if `True`, return the generic CCD response curve that we have on hand. If `False`, assume a constant throughput.

Reply:
Okay, I agree with this. Though I am curious what the controversy is..!